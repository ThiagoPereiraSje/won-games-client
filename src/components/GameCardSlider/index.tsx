import {
  ArrowBackIos as ArrowLeft,
  ArrowForwardIos as ArrowRight
} from 'styled-icons/material-outlined'
import GameCard, { GameCardProps } from 'components/GameCard'
import Slider, { SliderSettings } from 'components/Slider'
import * as S from './styles'

export type GameCardSliderColors = 'white' | 'black'

export type GameCardSliderProps = {
  items: GameCardProps[]
  color?: GameCardSliderColors
}

const settings: SliderSettings = {
  dots: false,
  infinite: false,
  slidesToShow: 4,
  slidesToScroll: 1,
  lazyLoad: 'ondemand',
  prevArrow: <ArrowLeft aria-label="previous games" />,
  nextArrow: <ArrowRight aria-label="next games" />,

  responsive: [
    {
      breakpoint: 1375,
      settings: {
        arrows: true,
        slidesToShow: 4.2
      }
    },

    {
      breakpoint: 1024,
      settings: {
        arrows: false,
        slidesToShow: 3.2
      }
    },

    {
      breakpoint: 768,
      settings: {
        arrows: false,
        slidesToShow: 2.2
      }
    },

    {
      breakpoint: 570,
      settings: {
        arrows: false,
        slidesToShow: 1.2
      }
    },

    {
      breakpoint: 375,
      settings: {
        arrows: false,
        slidesToShow: 1.1
      }
    }
  ]
}

const GameCardSlider = ({ items, color = 'white' }: GameCardSliderProps) => {
  return (
    <S.Wrapper color={color}>
      <Slider settings={settings}>
        {items.map((item, index) => (
          <GameCard key={index} {...item} />
        ))}
      </Slider>
    </S.Wrapper>
  )
}

export default GameCardSlider
