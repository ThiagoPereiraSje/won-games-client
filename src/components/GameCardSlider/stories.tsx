import { ComponentMeta, ComponentStory } from '@storybook/react'
import { GameCardProps } from 'components/GameCard'
import GameCardSlider from '.'

const items: GameCardProps[] = [
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x141',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x142',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x143',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x144',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x145',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x145',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  }
]

export default {
  title: 'GameCardSlider',
  component: GameCardSlider,
  args: {
    items
  },
  parameters: {
    layout: 'fullscreen',
    backgrounds: {
      default: 'dark'
    }
  }
} as ComponentMeta<typeof GameCardSlider>

export const Basic: ComponentStory<typeof GameCardSlider> = (args) => (
  <div style={{ maxWidth: '130rem', margin: '0 auto' }}>
    <GameCardSlider {...args} />
  </div>
)
