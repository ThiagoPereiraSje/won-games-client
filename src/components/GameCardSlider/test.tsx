import { render, screen } from '@testing-library/react'
import GameCardSlider from '.'

describe('<GameCardSlider />', () => {
  it('should render the heading', () => {
    render(<GameCardSlider />)

    const element = screen.getByRole('heading', { name: /GameCardSlider/i })

    expect(element).toBeInTheDocument()
  })

  it('should render correctly', () => {
    const { container } = render(<GameCardSlider />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
