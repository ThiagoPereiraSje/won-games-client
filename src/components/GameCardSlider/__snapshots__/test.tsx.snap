// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameCardSlider /> should render correctly 1`] = `
<main
  class=""
>
  <h1>
    Hello GameCardSlider
  </h1>
</main>
`;
